# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/util/APZEventState.h
# Commit: d9c5bf13b19a
# Full Hash: d9c5bf13b19a73883abad0a4e61b234aee446079
# Author: Dan Robertson <dan@dlrobertson.com>
# Date: 2023-10-06 15:27:33
# Description:
#   Bug 1856716 - Ensure that we hold a strong reference to the APZ event state. r=hiro
#   
#   - Ensure that we hold a strong reference to the APZ event state before
#      calling ProcessSingleTap, as ProcessSingleTap fires events.
#    - Mark ProcessSingleTap as MOZ_CAN_RUN_SCRIPT since it can fire events.
# ==============================================================================

diff -r 5c2d783060b0 -r d9c5bf13b19a gfx/layers/apz/util/APZEventState.h
--- a/gfx/layers/apz/util/APZEventState.h	Thu Oct 05 22:18:54 2023 +0000
+++ b/gfx/layers/apz/util/APZEventState.h	Thu Oct 05 22:34:32 2023 +0000
@@ -57,6 +57,7 @@
 
   NS_INLINE_DECL_REFCOUNTING(APZEventState);
 
+  MOZ_CAN_RUN_SCRIPT
   void ProcessSingleTap(const CSSPoint& aPoint,
                         const CSSToLayoutDeviceScale& aScale,
                         Modifiers aModifiers, int32_t aClickCount,